tools/pastel-utility/cmd/init.go

```diff
@@ -0,0 +1,307 @@
+			out, err := os.Create(zksnarkParamsPath)
+			if err != nil {
+				log.WithContext(ctx).WithError(err).Errorf("Error creating file: %s\n", zksnarkParamsPath)
+				return errors.Errorf("Failed to create file: %v \n", err)
+			}
+
+			log.WithContext(ctx).Infof("downloading: %s", zksnarkParamsPath)
+			resp, err := http.Get(configs.ZksnarkParamsURL + zksnarkParamsName)
+			if err != nil {
+				out.Close()
+				log.WithContext(ctx).WithError(err).Errorf("Error downloading file: %s\n", configs.ZksnarkParamsURL+zksnarkParamsName)
+				return errors.Errorf("Failed to download: %v \n", err)
+			}
+
+			_, err = io.Copy(out, resp.Body)
+			if err != nil {
+				out.Close()
+				resp.Body.Close()
+				return err
+			}
+
+			out.Close()
+			resp.Body.Close()
+		} else {
+			log.WithContext(ctx).Infof("zksnark file %s already exists and force set to %v", zksnarkParamsPath, force)
+		}
+	}
+	log.WithContext(ctx).Info("ZkSnark params download completed.\n")
+	return nil
+}
+
+// checkLocalAndRouterFirewalls checks local and router firewalls and suggest what to open
+func checkLocalAndRouterFirewalls(ctx context.Context, requiredPorts []string) error {
+	baseURL := "https://portchecker.com?q=" + strings.Join(requiredPorts[:], ",")
+	// resp, err := http.Get(baseURL)
+	// if err != nil {
+	// 	log.WithContext(ctx).WithError(err).Errorf("Error requesting url\n")
+	// 	return errors.Errorf("Failed to request port url %v \n", err)
+	// }
+	// defer resp.Body.Close()
+	// ok := resp.StatusCode == http.StatusOK
+	ok := true
+	if ok {
+		log.WithContext(ctx).Info("Your ports {} are opened and can be accessed by other PAstel nodes! ", baseURL)
+	} else {
+		log.WithContext(ctx).Info("Your ports {} are NOT opened and can NOT be accessed by other Pastel nodes!\n Please open this ports in your router firewall and in {} firewall", baseURL, "func_to_check_OS()")
+	}
+	return nil
+}
+
+// runWalletSubCommand runs wallet subcommand
+func runWalletSubCommand(ctx context.Context, config *configs.Config) error {
+	log.WithContext(ctx).Infoln("Wallet Node Start")
+	defer log.WithContext(ctx).Infoln("Wallet Node End")
+
+	log.WithContext(ctx).Infof("Config: ")
+	log.WithContext(ctx).Infof(config.String())
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	sys.RegisterInterruptHandler(cancel, func() {
+		log.WithContext(ctx).Info("Interrupt signal received. Gracefully shutting down...")
+		os.Exit(0)
+	})
+
+	walletNodePath := filepath.Join(config.WorkDir, "walletnode")
+
+	if _, err := os.Stat(walletNodePath); os.IsNotExist(err) || config.Init.Force {
+		if err := os.MkdirAll(walletNodePath, os.ModePerm); err != nil {
+			return errors.Errorf("could not create walletnode path %s, %v", walletNodePath, err)
+		}
+	}
+
+	v := viper.New()
+	v.SetConfigType("json")
+	if err := v.ReadConfig(bytes.NewBufferString(config.Init.String())); err != nil {
+		return err
+	}
+
+	walletConfigPath := fmt.Sprintf("%s/wallet.env", walletNodePath)
```

Comment:
```suggestion
	walletConfigPath := filepath.Join(walletNodePath, "walletnode.yml")
```
1. Do you realize that windows OS has `\` back slash? Use `fmt.Sprintf` to concat paths is silly
2. `wallet.env` this is wrong filename! Who did tell you that config name of `walletnode` is `wallet.env` ? Moreover you are use `v.SetConfigType("json")`. It's weird 
3. Config is not needed to be placed in `work-dir`